amazon_s3_download_block.py

```diff
@@ -1,22 +1,47 @@
-from nio.properties import (VersionProperty, StringProperty,
+import boto3
+
+from nio.block.base import Block
+from nio.signal.base import Signal
+from nio.properties import (VersionProperty, StringProperty, PropertyHolder,
                             ObjectProperty)
-from .amazon_base_block import AmazonBase, AWSCreds
-from nio.util.discovery import discoverable
 
 
-@discoverable
-class AmazonS3(AmazonBase):
+class AWSCreds(PropertyHolder):
+    aws_access_key_id = StringProperty(
+        title="Access Key ID", default="", allow_none=False)
+    aws_secret_access_key = StringProperty(
+        title="Secret Access Key", default="", allow_none=False)
+    aws_session_token = StringProperty(
+        title="Session Token", default="", allow_none=False)
+
+
+class AmazonS3Download(Block):
 
     version = VersionProperty("1.0.0")
     creds = ObjectProperty(
         AWSCreds, title="AWS Credentials", default=AWSCreds())
 
-    # Bucket in which file is stored
-    bucket_name = StringProperty(title="Bucket Name", default="bucket")
-    # Name of file in bucket
-    key = StringProperty(title="S3 File Key", default="")
-    # File to write to on local machine
-    file_name = StringProperty(title="S3 File Name", default="")
+    # S3 bucket to upload to
+    bucket_name = StringProperty(
+        title="Bucket Name", default="{{ $bucket_name }}")
+    # What to name file in S3 bucket
+    key = StringProperty(
+        title="S3 File Key", default="{{ $key }}")
+    # Path to file on local machine
+    file_name = StringProperty(
```

Comment:
If the above "key" property is the filename to get, this should be a `FileProperty` that is a path to store the DL'ed file on the machine. The title is also "File to upload"